client/landing/stepper/utils/path.ts

```diff
@@ -54,20 +58,11 @@ export const useLoginUrl = ( params: {
 	const loginPath =
 		locale && locale !== 'en' ? `/start/account/user/${ locale }` : `/start/account/user`;
 
-	const nonEmptyQueryParameters = Object.entries( params )
-		.filter( ( [ , value ] ) => value )
-		.map( ( [ key, value ] ) => {
-			switch ( key ) {
-				case 'redirectTo':
-					return [ 'redirect_to', value ];
-				case 'flowName':
-					return [ 'variationName', value ];
-				default:
-					return [ key, value ];
-			}
-		} );
+	const nonEmptyQueryParameters = {
+		...( flowName && { variationName: flowName } ),
+		...( redirectTo && { redirect_to: redirectTo } ),
+		...( pageTitle && { pageTitle } ),
```

Comment:
Let's try and trace `addQueryArgs` down to see if `undefined` or empty values will make it into the query string. If not, then we may not need the conditional spread either i.e. just pass them along:

```
addQueryArgs( loginPath, {
    variationName: flowName,
    redirect_to: redirectTo,
    pageTitle,
} )
```

Reply:
Good point! [Looks like undefined values are ignored ](https://github.com/WordPress/gutenberg/blob/96c6cfe8da3f6df46b16aa99f99b34aa8d0e2e49/packages/url/src/build-query-string.js#L44-L48)